core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -444,7 +444,9 @@ class CollectionAccumulator[T] extends AccumulatorV2[T, java.util.List[T]] {
 
   override def copy(): CollectionAccumulator[T] = {
     val newAcc = new CollectionAccumulator[T]
-    newAcc._list.addAll(_list)
+    _list.synchronized {
```

Comment:
@mridulm , I don't think that your statement is correct. `SynchronizedCollection` is protected by
`         final Object mutex;     // Object on which to synchronize
`

If you look at the constructor you will see that
`            mutex = this;
`

Does that make sense now?